### PR TITLE
Add extensions for model DocBlocks

### PR DESCRIPTION
**Pull request by Miljoen**
## Summary
Add ability to customize the model DocBlocks by registering invokable custom model extensions.

Before completing the checklist, is there any chance of this feature being merged in the future?
It will help us a lot for generating complete model DocBlocks.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

### Checklist
- [ ] Existing tests have been adapted and/or new tests have been added
- [ ] Add a CHANGELOG.md entry
- [ ] Update the README.md
- [ ] Code style has been fixed via `composer fix-style`

## Usage
```php
    /*
    |--------------------------------------------------------------------------
    | Model extensions
    |--------------------------------------------------------------------------
    |
    | Add ability to write invokable extensions for model DocBlocks.
    |
    */
    'model_extensions' => [
        \App\Support\IdeHelper\Paperclip::class,
    ],
```

```php
class Paperclip
{
    public function __invoke(Model $model, ModelsCommand $modelsCommand): void
    {
        if (! $model instanceof AttachableInterface) {
            return;
        }

        foreach ($model->getAttachedFiles() as $index => $attachment) {
            $modelsCommand->setProperty($index, Attachment::class, true, true, '', true);
        }
    }
}
```

### CODE REVIEW

**Comment by mfn** [comment]
It ticks the "looks useful and has a simple implementation"-checkmark for me and it seems on a conceptional level it competes with https://github.com/barryvdh/laravel-ide-helper/pull/945 , where I had _quite_ some reservation.

They did not really change but… 

People seem to love / require this feature 😏 

Funnily, _before_ I checked for the other PR I was already wondering if it isn't better to move it to the end, to be able to to see the final state and mutate it:

image

Which the other PR in fact does.

---
TL;DR: from a technical point I prefer the other PR:
- "hook" is at the end"
- we've an interface (which I prefer, `__invoke` seems to implicit for no good technical reason?)


well and the other PR actually seems finished with tests and we just failed to gather final feedback 😞

**Comment by Miljoen** [comment]
@mfn Thank you for the feedback.
My bad for not checking out the outstanding PRs, like #945.
That seems to do just what the people in this PR need.

So with that one being close to being merged, I see no more reason for this PR at this moment.

**Comment by mfn** [comment]
FTR I approved the other PR https://github.com/barryvdh/laravel-ide-helper/pull/945#pullrequestreview-610817237

**Comment by Miljoen** [comment]
With #945 merged, this MR can be closed! Thank you.

**Comment by mfn** [comment]
Oh right, still thanks for your PR @Miljoen !
